packages/@aws-cdk/aws-appmesh/README.md

```diff
@@ -1,7 +1,194 @@
+
+```typescript
+const mesh = new Mesh(this, 'AppMesh', {
+  name: props.appMeshName,
+  meshSpec: {
+    egressFilter: appmesh.MeshFilterType.Allow_All,
+  },
+});
+```
+
+## Exporting the Mesh
+
+The `Mesh` can also be exported to reference in other `Stacks`.
+
+```typescript
+const exportedMesh = mesh.export();
+
+const STACK_2 = new cdk.Stack(app, 'stack-2');
+const importedMesh = appmesh.Mesh.import(STACK_2, 'imported-mesh', exportedMesh);
+```
+
+## Adding VirtualRouters
+
+The `Mesh` needs `VirtualRouters` as logical units to route to `VirtualNodes`.
+
+Virtual routers handle traffic for one or more virtual services within your mesh. After you create a virtual router, you can create and associate routes for your virtual router that direct incoming requests to different virtual nodes.
+
+```typescript
+const router = mesh.addVirtualRouter('router', {
+  portMappings: [
+    {
+      port: 8081,
+      protocol: appmesh.Protocol.HTTP,
+    },
+  ],
+});
+```
+
+The listener protocol can be either `HTTP` or `TCP`.
+
+## Adding VirtualService
+
+A virtual service is an abstraction of a real service that is provided by a virtual node directly or indirectly by means of a virtual router. Dependent services call your virtual service by its virtualServiceName, and those requests are routed to the virtual node or virtual router that is specified as the provider for the virtual service.
+
+We recommend that you use the service discovery name of the real service that you're targeting (such as `my-service.default.svc.cluster.local`).
+
+For the provider property:
+
+* If you want the virtual service to spread traffic across multiple virtual nodes, select Virtual router and then choose the virtual router to use from the drop-down menu.
+* If you want the virtual service to reach a virtual node directly, without a virtual router, select Virtual node and then choose the virtual node to use from the drop-down menu.
+* If you don't want the virtual service to route traffic at this time (for example, if your virtual nodes or virtual router doesn't exist yet), leave blank. You can update the provider for this virtual service later.
+
+ * Adds a virtual router as the provider
+  
+```typescript
+mesh.addVirtualService('virtual-service', {
+  virtualRouterName: router.virtualRouterName,
+  virtualServiceName: 'my-service.default.svc.cluster.local',
+});
+```
+
+* Adds a virtual node as the provider
+
+```typescript
+mesh.addVirtualService('virtual-service', {
+  virtualNodeName: node.virtualNodeName,
+  virtualServiceName: `ratings.${infra.namespace.namespaceName}`,
+});
+```
+
+**Note** that only one must of `virtualNodeName` or `virtualRouterName` must be chosen.
+
+## Adding a VirtualNode
+
+A `virtual node` acts as a logical pointer to a particular task group, such as an Amazon ECS service or a Kubernetes deployment.
+
+![vn](https://docs.aws.amazon.com/app-mesh/latest/userguide/images/virtual_node.png)
```

Comment:
Please use a better alternate text so this is screen-reader friendly.

Reply:
You're so right on this one, need to think through accessibility a much more, it'll be fixed on the next push.